config/check_std_isnan.cxx

```diff
@@ -0,0 +1,8 @@
+#include <cmath>
+int main()
+{
+    // Some old versions of libc++ do not provide isnan() for integers,
+    //  in which case this will fail to compile.
+    std::isnan(int(0));
```

Comment:
Implicit conversion doesn't happen because `enable_if`is used.  Here's the `isnan` implementation in my outdated copy of `libc++`:

```
template <class _A1>
inline _LIBCPP_INLINE_VISIBILITY
typename std::enable_if<std::is_floating_point<_A1>::value, bool>::type
isnan(_A1 __x) _NOEXCEPT
{
    return __libcpp_isnan(__x);
}
```